mirror-godot-app/scripts/autoload/zone/server.gd

```diff
@@ -272,31 +273,41 @@ func _handle_zone_socket_error(request: Dictionary) -> void:
 	push_error(error_msg, " ", request)
 	print(error_msg)
 
-
+var _loaded_pages = 0
+var _queued_assets = []
 func _on_zone_socket_space_object_page_received(space_objects_page: Dictionary) -> void:
 	var page = space_objects_page.get("page", 1)
 	var total_pages = space_objects_page.get("totalPage", 1)
 	var space_objects_arr = space_objects_page.get("result", [])
-	for obj in space_objects_arr:
-		# if we already have the asset in memory, no need to get it again
-		if not Net.asset_client.get_asset_json(obj["asset"]).is_empty():
-			continue
-		Net.zone_socket.queue_download_asset(obj["asset"])
+	_loaded_pages +=1
+	print("Loaded page: ", page, " time: ", Time.get_datetime_string_from_system())
+	print("Total pages loaded: ", _loaded_pages, " total pages ", total_pages)
 
-	for space_obj in space_objects_arr:
-		#assert(not space_obj.has("receipt"), "May happen with old spaces, but should not happen with new spaces.")
-		if space_obj.has("receipt"):
-			space_obj["creator"] = space_obj["receipt"].get("created_by_user", "")
-			space_obj.erase("receipt")
 	Zone.space_objects.append_array(space_objects_arr)
 	# if we've reached the final page, we're done getting all the data
```

Comment:
Explain why the algorithm is better when it fetches more pages in the same moment in a comment, before it was sequentially executing instructions, which is slow. Instead put all the requests in a block and get all the data back much sooner